src/node_crypto.cc

```diff
@@ -212,6 +212,115 @@ static int NoPasswordCallback(char* buf, int size, int rwflag, void* u) {
 }
 
 
+// namespace node::crypto::error
+namespace error {
+Local<Object> Decorate(Environment* env, Local<Object> obj,
+                             unsigned long err) {  // NOLINT(runtime/int)
+  if (err == 0) return obj;  // No decoration possible.
+
+  const char* ls = ERR_lib_error_string(err);
+  const char* fs = ERR_func_error_string(err);
+  const char* rs = ERR_reason_error_string(err);
+
+  Isolate* isolate = env->isolate();
+  Local<Context> context = isolate->GetCurrentContext();
+
+  if (ls != nullptr)
+    if (obj->Set(context, env->library_string(),
+        OneByteString(isolate, ls)).IsNothing())
+      return obj;
+  if (fs != nullptr)
+    if (obj->Set(context, env->function_string(),
+        OneByteString(isolate, fs)).IsNothing())
+      return obj;
+  if (rs != nullptr) {
+    if (obj->Set(context, env->reason_string(),
+        OneByteString(isolate, rs)).IsNothing())
+      return obj;
+
+    // SSL has no API to recover the error name from the number, so we
+    // transform reason strings like "this error" to "ERR_SSL_THIS_ERROR",
+    // which ends up being close to the original error macro name.
+    std::string reason(rs);
+
+    for (auto& c : reason) {
+      if (c == ' ')
+        c = '_';
+      else
+        c = ToUpper(c);
+    }
+
+#define OSSL_ERROR_CODES_MAP(V) \
+    V(SYS) \
+    V(BN) \
+    V(RSA) \
+    V(DH) \
+    V(EVP) \
+    V(BUF) \
+    V(OBJ) \
+    V(PEM) \
+    V(DSA) \
+    V(X509) \
+    V(ASN1) \
+    V(CONF) \
+    V(CRYPTO) \
+    V(EC) \
+    V(SSL) \
+    V(BIO) \
+    V(PKCS7) \
+    V(X509V3) \
+    V(PKCS12) \
+    V(RAND) \
+    V(DSO) \
+    V(ENGINE) \
+    V(OCSP) \
+    V(UI) \
+    V(COMP) \
+    V(ECDSA) \
+    V(ECDH) \
+    V(OSSL_STORE) \
+    V(FIPS) \
+    V(CMS) \
+    V(TS) \
+    V(HMAC) \
+    V(CT) \
+    V(ASYNC) \
+    V(KDF) \
+    V(SM2) \
+    V(USER) \
```

Comment:
It's sorta customary to line up the backslashes in the node base. I guess I should have mentioned that during my last review. Mea culpa!

Reply:
at column 79, or just far enough to be vertically aligned?

Reply:
found other examples, aligned.